src/Orleans/Streams/PersistentStreams/StreamQueueBalancerType.cs

```diff
@@ -39,5 +39,10 @@ public enum StreamQueueBalancerType
         /// That is, it does not rebalance queues based on dymanic changes in the cluster Membership.
         /// </summary>
         StaticClusterConfigDeploymentBalancer,
+
+        /// <summary>
+        /// Use user configured custom queue balancer
+        /// </summary>
+        CustomBalancer
```

Comment:
@jason-bragg  has similar concern on that configuration is a bit redundant. He suggested something which convinced me, and I think you will like it. 

He suggested remove this type, and make StreamQueueBalancerType.cs a list of static string, which list the build in queue balancers. so the runtime can resolve the queue balancer using named service. And when user want to inject their own queue balancer, they can configure PersistenStreamProviderConfig.BalancerType to be their custom queue balancer type by giving it a name. And we can remove the other QueueBalancerFactoryName config . 

I think it is similar to what you suggested. Hope this make sense, or you will see in my PR update ;)